charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -150,7 +150,7 @@ spec:
           mountPath: /consul-bin
         {{- end }}
         {{- if .Values.global.tls.enabled }}
-        {{- if .Values.global.tls.enableAutoEncrypt }}
+        {{- if and .Values.client.enabled .Values.global.tls.enableAutoEncrypt }}
         - name: consul-auto-encrypt-ca-cert
         {{- else }}
         - name: consul-ca-cert
```

Comment:
So, like mentioned below (in the init portion), if we _do_ want to fix the behavior with `useSystemRoots` in this PR (or I think we can save it for a subsequent PR), I believe we'll have to drop this from being referenced if and only if we have: 1) `externalServers.useSystemRoots` set to `true` and 2) `client.enabled` set to `false`. In that case, I don't believe we'll actually ever create the `consul-ca-cert` secret.

Without those changes to the conditional mount we get the following issue on install (surfaced in a recent HCP configuration when we were trying to update the learn guide and dropping what we thought were unused options):

> │ Error: Deployment.apps "consul-api-gateway-controller" is invalid: [spec.template.spec.containers[0].volumeMounts[1].name: Not found: "consul-ca-cert", spec.template.spec.initContainers[1].volumeMounts[1].name: Not found: "consul-ca-cert"]
│

Reply:
I fixed it for the init container and I need to fix it for the _container_ container.

Reply:
Fixed in both places. I'll do more testing to confirm in the morning.